_episodes/10-prog.md

```diff
@@ -180,6 +180,87 @@ and translates any special characters in the values
 into their escaped equivalents
 so that they are safe to use.
 
+We can also use `sqlite3`'s cursor to make changes to our database,
+such as inserting a new name.
+For instance, we can define a new function called `add_name` like so:
+
+~~~
+def add_name(database_file, new_person):
+    query = "INSERT INTO Person VALUES (?, ?, ?);"
+
+    connection = sqlite3.connect(database_file)
+    cursor = connection.cursor()
+    cursor.execute(query, list(new_person))
+    cursor.close()
+    connection.close()
+
+
+def get_name(database_file, person_id):
+    query = "SELECT personal || ' ' || family FROM Person WHERE id=?;"
+
+    connection = sqlite3.connect(database_file)
+    cursor = connection.cursor()
+    cursor.execute(query, [person_id])
+    results = cursor.fetchall()
+    cursor.close()
+    connection.close()
+
+    return results[0][0]
+
+# Insert a new name
+add_name('survey.db', ('barrett', 'Mary', 'Barrett'))
+# Check it exists
+print("Full name for barrett:", get_name('survey.db', 'barrett'))
+~~~
+{: .python}
+~~~
+IndexError: list index out of range
+~~~
+{: .output}
+
+Note that in versions of sqlite3 >= 2.5, the `get_name` function described
+above will fail with an `IndexError: list index out of range`,
+even though we added Mary's
+entry into the table using `add_name`.
+This is because we must perform a `connection.commit()` before closing
+the connection, in order to save our changes to the database.
+
+~~~
+def add_name(database_file, new_person):
+    query = "INSERT INTO Person VALUES" + repr(person_details) + ";"
```

Comment:
@matty-jones, Where is person_details coming from ?

Reply:
Hi Henry, thanks for weighing in. This was a typo and has been fixed in a subsequent commit.

Reply:
Hi @matty-jones I have tested this and all seems fine. I am waiting for the change from `repr(person_details)` > `repr(new_person)` or keep the version above and include a commit statement 

```python
    query = "INSERT INTO Person VALUES (?, ?, ?);"

    connection = sqlite3.connect(database_file)
    cursor = connection.cursor()
    cursor.execute(query, list(new_person))
```
Also it would look nice if we added the import sqlite3 statement with a line after, for these two added codes
```python 
import sqlite3

def add_name(database_file, new_person):
    query = 

```

Reply:
Hi @henrykironde,

That repr() was completely removed in https://github.com/swcarpentry/sql-novice-survey/pull/230/commits/6e9e547af167d2fbf8eb9a12a3be66cad0e09ed6 based on @remram44 's suggestions.

There should be 3 commits for this pull request, are you sure that you are reviewing the latest commit? Is there something that I must do to update the pull request?

RE: the import sqlite3: I agree with your suggestions and I will update this. Thank you!